aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/OnErrorConsumer.kt

```diff
@@ -24,7 +24,7 @@ class OnErrorConsumer<T: Model>( val emitter: SingleEmitter<PaginatedResult<Mode
                result
             )
         } else {
-            retryHandler.retry(emitter,appSync,request, onResponse, this)
+            retryHandler.retry(RetryHandler(emitter, appSync, request, onResponse, this))
```

Comment:
Since the `RetryHandler` class is really only two lines of code, maybe just put the implementation inline, like this?

```suggestion
            retryHandler.retry({
               val cancelable = appSync.sync(request, onResponse, this)
               emitter.setDisposable(AmplifyDisposables.fromCancelable(cancelable))
        })
```

Reply:
Would this work with Java? I need to convet this class back to java.

Reply:
I can create an inline implementation of the interface in Java.